app/lib/lnd/util.js

```diff
@@ -163,15 +164,17 @@ export const waitForFile = (filepath, timeout = 1000) => {
 
   // Promise B resolves when the file has been found.
   let promiseB = new Promise(resolve => {
-    let intervalId = setInterval(() => {
+    let intervalId = setInterval(async () => {
       mainLog.debug('waiting for file: %s', filepath)
-      if (!fs.existsSync(filepath)) {
+      try {
+        await stat(filepath)
+        mainLog.debug('found file: %s', filepath)
+        clearInterval(intervalId)
```

Comment:
Probably better to clear interval and timeout before `await stat(filepath)` incase it throws. Or even better add finally block

Reply:
We want to clear the timers in one of 2 different places. Either here after the file has been successfully found. Or, in the other promise when the timeout has been reached. We don't want to remove the timers in the case of a throw, since in that instance we want to keep rechecking for the file, scheduled by the interval timer.